crest/Assets/Crest/Crest/Scripts/Helpers/OceanRendererLifeCycleHelper.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Crest Ocean System
+
+// This file is subject to the MIT License as seen in the root of this folder structure (LICENSE)
+
+using UnityEngine;
+
+namespace Crest
+{
+    /// <summary>
+    /// Enables/disables a component based on the OceanRenderer's life cycle.
+    /// </summary>
+    public class OceanRendererLifeCycleHelper
+    {
+        // The mono behaviour we want to track the enabled/disabled preference.
+        readonly MonoBehaviour _monoBehaviour;
+        // Store the enabled/disabled checkbox set by the user so we can respect it.
+        bool _enabledUserPreference;
+        // Whether the state change came from the user or from the ocean life cycle event.
+        bool _isStateChangeFromUser = true;
+
+        public OceanRendererLifeCycleHelper(MonoBehaviour monoBehaviour)
+        {
+            _monoBehaviour = monoBehaviour;
+            _enabledUserPreference = _monoBehaviour.enabled;
+
+            OceanRenderer.OnOceanRendererEnabled -= OnOceanRendererEnabled;
+            OceanRenderer.OnOceanRendererEnabled += OnOceanRendererEnabled;
+            OceanRenderer.OnOceanRendererDisabled -= OnOceanRendererDisabled;
+            OceanRenderer.OnOceanRendererDisabled += OnOceanRendererDisabled;
+        }
+
+        /// <summary>
+        /// Needs to be called to clean up resources. Destructors do not work when domain reload is disabled.
+        /// </summary>
+        public void OnDestroy()
+        {
+            OceanRenderer.OnOceanRendererEnabled -= OnOceanRendererEnabled;
+            OceanRenderer.OnOceanRendererDisabled -= OnOceanRendererDisabled;
+        }
+
+        /// <summary>
+        /// Sets enabled based on OceanRenderer's life cycle. Callee must return early if false is returned.
+        /// </summary>
+        public bool OnEnable()
+        {
+            if (_isStateChangeFromUser)
```

Comment:
maybe another comment - i think i know what the bool represents, but i cant quite get my head around how it works / how it has the right value at this time

Reply:
I added a comment. We set it to false whenever we know it won't be the user.